extension/core_functions/scalar/list/list_filter.cpp

```diff
@@ -30,8 +30,9 @@ static unique_ptr<FunctionData> ListFilterBind(ClientContext &context, ScalarFun
 	return LambdaFunctions::ListLambdaBind(context, bound_function, arguments, has_index);
 }
 
-static LogicalType ListFilterBindLambda(const idx_t parameter_idx, const LogicalType &list_child_type) {
-	return LambdaFunctions::BindBinaryLambda(parameter_idx, list_child_type);
+static LogicalType ListFilterBindLambda(const idx_t parameter_idx, ClientContext &context,
```

Comment:
Nit: let's pass the client context as the first argument. Maybe context, types, parameter index?